File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/models/Debt.java
package com.androidcollider.easyfin.common.models;

import java.io.Serializable;

import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

/**
 * @author Ihor Bilous
 */

@Getter
@Builder
@ToString
@EqualsAndHashCode(exclude = {"type", "idAccount", "name", "accountName", "currency", "amountCurrent", "amountAll", "accountAmount", "date"})
public class Debt implements Serializable {

    @Setter
    private int id;
    private int type;
    private int idAccount;
    private String name;
    private String accountName;
    private String currency;
    @Setter
    private double amountCurrent;
    @Setter
    private double amountAll;
    private double accountAmount;
    private long date;
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/common/models/Debt.kt
package com.androidcollider.easyfin.common.models

import java.io.Serializable

/**
 * @author Ihor Bilous
 */
class Debt : Serializable {
    var id = 0
    var type = 0
    var idAccount = 0
    var name: String? = null
    var accountName: String? = null
    var currency: String? = null
    var amountCurrent = 0.0
    var amountAll = 0.0
    var accountAmount = 0.0
    var date: Long = 0

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        other as Debt
        if (id != other.id) return false
        return true
    }

    override fun hashCode(): Int {
        return id
    }
}